### PR TITLE
v6 (major release): Use a newer Rust

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Newer rust: `nightly-2023-12-11`.
 - Handle `sc-config.toml`, as well.

Also see:
 - https://github.com/multiversx/mx-sdk-py-cli/issues/383
 - https://t.me/MultiversXDevelopers/122082
 - https://github.com/multiversx/mx-docs/pull/788
 - https://github.com/multiversx/mx-reproducible-contract-build-example-sc/pull/21

### CODE REVIEW

File: integration_tests/previous_builds.py [hunk 1]
```diff
@@ -48,17 +48,32 @@ def __init__(self, name: str,
...
+            "multisig-full": "9eed9c35113209fc69631cf29aac6e81f0e331132bf6e46198e679259075ad49",
+            "multisig-view": "3993cf3fb5cd18102e2b8946ea1997f6f1cc512537f453265ba1afd7378fc0c6",
             "lottery-esdt": "e06b1a5c7fb71181a79e9be6b86d8ad154e5c2def4da6d2f0aa5266163823291"
         },
-        docker_image="multiversx/sdk-rust-contract-builder:v5.3.0"
+        docker_image="multiversx/sdk-rust-contract-builder:v5.4.0"
+    ),
+    PreviousBuild(
+        name="a.4",
+        project_archive_url="https://github.com/multiversx/mx-reproducible-contract-build-example-sc/archive/refs/tags/v0.5.0-beta.0.zip",
```
Comment (andreibancioiu) [inline integration_tests/previous_builds.py:66]: For these tests, proper tags will be added at a later time.

File: integration_tests/previous_builds.py [hunk 2]
```diff
@@ -48,17 +48,32 @@ def __init__(self, name: str,
...
+        packaged_src_url=None,
+        contract_name=None,
+        expected_code_hashes={
+            "adder": "384b680df7a95ebceca02ffb3e760a2fc288dea1b802685ef15df22ae88ba15b",
+            "multisig": "d1453017d1fcac43f3b54c390b112b37af38ae840a2464d8ff68e3981da9972d",
+            "multisig-full": "e7fd6d118639e4b4381b3a667435948cd70f1f06b6ef39e227bd88349f7e7979",
+            "multisig-view": "290c9b3e374dffa33649ed46bd0b626c66f933eff9437f11f3559372d7538f85",
+            "lottery-esdt": "d4d4b6d2d797749435a4127a12d5ea16b911d6783e00cbc9faf4bace7d655c7a"
+        },
+        docker_image="sdk-rust-contract-builder:next"
```
Comment (andreibancioiu) [inline integration_tests/previous_builds.py:77]: For these tests, proper tags will be added at a later time.

File: multiversx_sdk_rust_contract_builder/source_code.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def _is_source_code_file(path: Path) -> bool:
         return True
     if path.parent.name == "meta" and path.name == "Cargo.lock":
         return False
-    if path.name in ["Cargo.toml", "Cargo.lock", "multicontract.toml", CONTRACT_CONFIG_FILENAME]:
+    if path.name in ["Cargo.toml", "Cargo.lock", "multicontract.toml", "sc-config.toml", CONTRACT_CONFIG_FILENAME]:
```
Comment (andreibancioiu) [inline multiversx_sdk_rust_contract_builder/source_code.py:66]: Very important - we should apply this to LTS v5, as well.

File: integration_tests/test_project_folder_and_packaged_src_are_equivalent.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 def main(cli_args: List[str]):
     repository_url = "https://github.com/multiversx/mx-reproducible-contract-build-example-sc"
-    tag = "0.4.7-beta.1"
+    tag = "0.5.0-beta.0"
```
Comment (andreibancioiu) [inline integration_tests/test_project_folder_and_packaged_src_are_equivalent.py:12]: For these tests, proper tags will be added at a later time.

**Review by CostinCarabas — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**